src/daemon/control.rs

```diff
@@ -820,6 +820,36 @@ pub fn fetch_cosigs_signatures(
     Ok(())
 }
 
+/// This function estimates (conservatively) the size of the message
+/// for sending the fully-signed tx to the coordinator, returning
+/// if the size is smaller than NOISE_PLAINTEXT_MAX_SIZE
+pub fn check_spend_transaction_size(
+    revaultd: &RevaultD,
+    spend_tx: SpendTransaction,
+    n_deposit_outpoints: usize,
+) -> bool {
+    // Very rough estimate of the SetSpendTx message size
+    // Each deposit takes ~70 bytes (64 for txid + ":" + 1 for vout + some room for other chars
+    // such as commas)
+    // The remaining parameters take approximately 100 bytes
+    let overhead = n_deposit_outpoints * 70 + 100;
+    let tx_finalized = spend_tx.is_finalized();
+    let tx = spend_tx.into_psbt().extract_tx();
+    let mut final_tx_size = tx.get_size();
+
+    if !tx_finalized {
+        let max_satisfaction_weight = revaultd
+            .unvault_descriptor
+            .inner()
+            .max_satisfaction_weight()
+            .expect("Script must be satisfiable");
+        final_tx_size += max_satisfaction_weight * n_deposit_outpoints;
+    }
+
+    // We multiply final_tx_size by 2 as we're sending it in hex
+    return 2 * final_tx_size + overhead <= revault_net::noise::NOISE_PLAINTEXT_MAX_SIZE;
+}
```

Comment:
How about instead of the boutique computation (**untested**):
```diff
diff --git a/src/daemon/control.rs b/src/daemon/control.rs
index bc424e9..311eba0 100644
--- a/src/daemon/control.rs
+++ b/src/daemon/control.rs
@@ -823,19 +823,9 @@ pub fn fetch_cosigs_signatures(
 /// This function estimates (conservatively) the size of the message
 /// for sending the fully-signed tx to the coordinator, returning
 /// if the size is smaller than NOISE_PLAINTEXT_MAX_SIZE
-pub fn check_spend_transaction_size(
-    revaultd: &RevaultD,
-    spend_tx: SpendTransaction,
-    n_deposit_outpoints: usize,
-) -> bool {
-    // Very rough estimate of the SetSpendTx message size
-    // Each deposit takes ~70 bytes (64 for txid + ":" + 1 for vout + some room for other chars
-    // such as commas)
-    // The remaining parameters take approximately 100 bytes
-    let overhead = n_deposit_outpoints * 70 + 100;
+pub fn check_spend_transaction_size(revaultd: &RevaultD, spend_tx: SpendTransaction) -> bool {
     let tx_finalized = spend_tx.is_finalized();
-    let tx = spend_tx.into_psbt().extract_tx();
-    let mut final_tx_size = tx.get_size();
+    let mut tx = spend_tx.into_psbt().extract_tx();
 
     if !tx_finalized {
         let max_satisfaction_weight = revaultd
@@ -843,11 +833,19 @@ pub fn check_spend_transaction_size(
             .inner()
             .max_satisfaction_weight()
             .expect("Script must be satisfiable");
-        final_tx_size += max_satisfaction_weight * n_deposit_outpoints;
+        for input in tx.input.iter_mut() {
+            // It's not exact, but close enough
+            input.witness.push(vec![0; max_satisfaction_weight]);
+        }
     }
 
-    // We multiply final_tx_size by 2 as we're sending it in hex
-    return 2 * final_tx_size + overhead <= revault_net::noise::NOISE_PLAINTEXT_MAX_SIZE;
+    let deposit_outpoints: Vec<OutPoint> = tx.input.iter().map(|i| i.previous_output).collect();
+    let tx_hex = encode::serialize_hex(&tx);
+    let msg = serde_json::to_string(&serde_json::json!( {
+        "deposit_outpoints": deposit_outpoints,
+        "transaction": tx_hex,
+    })).expect("JSON created inline");
+    return msg.len() <= revault_net::noise::NOISE_PLAINTEXT_MAX_SIZE;
 }
 
 /// Sends the spend transaction for a certain outpoint to the coordinator
diff --git a/src/daemon/jsonrpc/api.rs b/src/daemon/jsonrpc/api.rs
index 67bad48..47dc2e6 100644
--- a/src/daemon/jsonrpc/api.rs
+++ b/src/daemon/jsonrpc/api.rs
@@ -983,7 +983,7 @@ impl RpcApi for RpcImpl {
             JsonRpcError::invalid_params(format!("Error while building spend transaction: {}", e))
         })?;
 
-        if !check_spend_transaction_size(&revaultd, tx_res.clone(), outpoints.len()) {
+        if !check_spend_transaction_size(&revaultd, tx_res.clone()) {
             return Err(JsonRpcError::invalid_params(format!(
                 "Spend transaction is too large, try spending less outpoints"
             )));
@@ -1178,13 +1178,7 @@ impl RpcApi for RpcImpl {
         })?;
 
         // Check that we can actually send the tx to the coordinator...
-        let deposit_outpoints: Vec<_> = spent_vaults
-            .values()
-            .map(|db_vault| db_vault.deposit_outpoint)
-            .collect();
-
-        if !check_spend_transaction_size(&revaultd, spend_tx.psbt.clone(), deposit_outpoints.len())
-        {
+        if !check_spend_transaction_size(&revaultd, spend_tx.psbt.clone()) {
             return Err(JsonRpcError::invalid_params(format!(
                 "Spend transaction is too large, try spending less outpoints"
             )));
@@ -1208,6 +1202,10 @@ impl RpcApi for RpcImpl {
         })?;
 
         // And then announce it to the Coordinator
+        let deposit_outpoints: Vec<_> = spent_vaults
+            .values()
+            .map(|db_vault| db_vault.deposit_outpoint)
+            .collect();
         announce_spend_transaction(&revaultd, finalized_spend, deposit_outpoints).map_err(|e| {
             JsonRpcError::invalid_params(format!(
                 "Communication error while announcing the Spend transaction: {}",
```